lib/trento/application/integration/discovery/payloads/cluster/crmmon_discovery_payload.ex

```diff
@@ -0,0 +1,264 @@
+      last_change
+      |> cast(attrs, [:time])
+      |> validate_required_fields([:time])
+    end
+  end
+
+  @required_fields [
+    :version,
+    :summary,
+    :resources,
+    :clones,
+    :node_history,
+    :node_attributes
+  ]
+  use Trento.Type
+
+  deftype do
+    field :version, :string
+
+    embeds_one :summary, Summary
+
+    embeds_many :resources, CrmmonResource
+
+    embeds_many :groups, CrmmonGroup, primary_key: false do
+      field :id, :string
+
+      embeds_many :primitives, Primitive
+      embeds_many :resources, CrmmonResource
+    end
+
+    embeds_many :clones, CrmmonClone, primary_key: false do
+      field :id, :string
+      field :failed, :boolean
+      field :unique, :boolean
+      field :managed, :boolean
+      field :multi_state, :boolean
+      field :failure_ignored, :boolean
+
+      embeds_many :resources, CrmmonResource
+    end
+
+    embeds_one :node_history, NodeHistory
+
+    embeds_one :node_attributes, NodeAttributes do
+      embeds_many :nodes, Node do
+        field :name, :string
+
+        embeds_many :attributes, Attribute do
+          field :name, :string
+          field :value, :string
+        end
+      end
+    end
+  end
+
+  def changeset(crmmon, attrs) do
+    filtered_attrs = transform_nil_lists(attrs)
+
+    crmmon
+    |> cast(filtered_attrs, [:version])
+    |> cast_embed(:summary)
+    |> cast_embed(:resources)
+    |> cast_embed(:groups, with: &groups_changeset/2)
+    |> cast_embed(:clones, with: &clones_changeset/2)
+    |> cast_embed(:node_history)
+    |> cast_embed(:node_attributes, with: &node_attributes_changeset/2)
+    |> validate_required_fields(@required_fields)
+  end
+
+  defp groups_changeset(groups, attrs) do
+    groups
+    |> cast(attrs, [:id])
+    |> cast_embed(:resources)
+    |> cast_embed(:primitives)
+    |> validate_required_fields([:id, :resources, :primitives])
+  end
+
+  defp clones_changeset(clones, attrs) do
+    clones
+    |> cast(attrs, [:id, :failed, :unique, :managed, :multi_state, :failure_ignored])
+    |> cast_embed(:resources)
+    |> validate_required_fields([
+      :id,
+      :failed,
+      :unique,
+      :managed,
+      :multi_state,
+      :failure_ignored,
+      :resources
+    ])
+  end
+
+  defp node_attributes_changeset(node_attributes, attrs) do
+    node_attributes
+    |> cast(attrs, [])
+    |> cast_embed(:nodes, with: &nodes_changeset/2)
+    |> validate_required_fields([:nodes])
+  end
+
+  defp nodes_changeset(nodes, attrs) do
+    nodes
+    |> cast(attrs, [:name])
+    |> cast_embed(:attributes, with: &attributes_changeset/2)
+    |> validate_required_fields([:name, :attributes])
+  end
+
+  defp attributes_changeset(attributes, attrs) do
+    attributes
+    |> cast(attrs, [:name, :value])
+    |> validate_required_fields([:name, :value])
+  end
+
+  defp transform_nil_lists(%{"groups" => groups, "clones" => clones} = attrs) do
+    attrs
+    |> Map.put("groups", ListHelper.to_list(groups))
```

Comment:
should we add here maybe the `resources` entry too? It might come as `null` if there is not resource configured